### ISSUE TITLE
java.lang.ClassCastException: class org.jooq.impl.ResultImpl cannot be cast to class java.sql.ResultSet when mocking Oracle SYS_REFCURSOR procedure

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11560]
Creating a procedure like this:

```sql
CREATE OR REPLACE PROCEDURE p_date_cursor (p_dates OUT SYS_REFCURSOR)
IS
BEGIN
  OPEN p_dates FOR
    SELECT
      NULL "d",
      NULL "ts"
    FROM dual
    UNION ALL
    SELECT
      CAST(TIMESTAMP '2015-01-01 15:30:00' AS DATE) AS "d",
           TIMESTAMP '2015-01-01 15:30:00'          AS "ts"
    FROM dual
    ORDER BY 1 NULLS FIRST;
END p_date_cursor;
```

And then mocking it like this:

```java
MockDataProvider p = c -> {
    var create = DSL.using(SQLDialect.ORACLE);
    var d = DSL.field("d", SQLDataType.DATE);
    var ts = DSL.field("ts", SQLDataType.TIMESTAMP);
    var cursorResult = create.newResult(d, ts);
    cursorResult.add(create.newRecord(d, ts).values(Date.valueOf("2000-01-01"), Timestamp.valueOf("2000-01-01 00:00:00")));
    var returnValue =  DSL.field("RETURN_VALUE", SQLDataType.RESULT);

    // Create new Result and set its single row value for returnValue to the cursor results.
    var outResult = create.newResult(returnValue);
    outResult.add(create.newRecord(returnValue).values((Result) cursorResult));

    return new MockResult[] {
        new MockResult(1, outResult)
    };
};

Routines.pDateCursor(new DefaultConfiguration().set(new MockConnection(p)).set(ORACLE));
```

Leads to the following exception:

```
Exception in thread "main" java.lang.ClassCastException: class org.jooq.impl.ResultImpl cannot be cast to class java.sql.ResultSet (org.jooq.impl.ResultImpl is in unnamed module of loader 'app'; java.sql.ResultSet is in module java.sql of loader 'platform')
	at org.jooq_3.15.0-SNAPSHOT.ORACLE.debug(Unknown Source)
	at org.jooq.impl.DefaultBinding$DefaultResultBinding.get0(DefaultBinding.java:3705)
	at org.jooq.impl.DefaultBinding$DefaultResultBinding.get0(DefaultBinding.java:1)
	at org.jooq.impl.DefaultBinding$AbstractBinding.get(DefaultBinding.java:927)
	at org.jooq.impl.AbstractRoutine.fetchOutParameter(AbstractRoutine.java:1229)
	at org.jooq.impl.AbstractRoutine.fetchOutParameters(AbstractRoutine.java:1199)
	at org.jooq.impl.AbstractRoutine.executeCallableStatement(AbstractRoutine.java:529)
	at org.jooq.impl.AbstractRoutine.execute(AbstractRoutine.java:362)
	at org.jooq.impl.Tools.attach(Tools.java:1212)
	at org.jooq.impl.AbstractRoutine.execute(AbstractRoutine.java:343)
	at org.jooq.test.oracle.generatedclasses.test.Routines.pDateCursor(Routines.java:1578)
```

----
See also:
https://groups.google.com/g/jooq-user/c/uU09EB5zblc

**Comment by lukaseder** [comment]
I vaguely remember this being a known limitation / missing feature, but cannot find the relevant issue.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.9 (https://github.com/jOOQ/jOOQ/issues/11561)
